aesd-char-driver/main.c

```diff
@@ -0,0 +1,202 @@
+/**
+ * @file aesdchar.c
+ * @brief Functions and data related to the AESD char driver implementation
+ *
+ * Based on the implementation of the "scull" device driver, found in
+ * Linux Device Drivers example code.
+ *
+ * @author Dan Walkes
+ * @date 2019-10-22
+ * @copyright Copyright (c) 2019
+ *
+ */
+
+#include <linux/module.h>
+#include <linux/init.h>
+#include <linux/printk.h>
+#include <linux/types.h>
+#include <linux/cdev.h>
+#include <linux/slab.h> // kmalloc
+#include <linux/fs.h> // file_operations
+#include "aesdchar.h"
+#include "aesd-circular-buffer.h"
+int aesd_major =   0; // use dynamic major
+int aesd_minor =   0;
+
+MODULE_AUTHOR("Ashwin Ravindra"); /** TODO: fill in your name **/
+MODULE_LICENSE("Dual BSD/GPL");
+
+struct aesd_dev aesd_device;
+char* final_buffptr = NULL;
```

Comment:
@ashwinhr186 suggest a refactor to put this inside of your aesd_dev structure, since your current implementation won't be safe with respect to locking.